include/deal.II/base/numbers.h

```diff
@@ -397,7 +397,7 @@ namespace internal
   template <typename T>
   struct NumberType
   {
-    static DEAL_II_CUDA_HOST_DEV T value (const T &t)
+    static DEAL_II_CUDA_HOST_DEV const T &value (const T &t)
```

Comment:
@Rombur If this PR gets merged as is, and `CUDA` breaks then please keep this change in mind!